.travis.yml

```diff
@@ -6,13 +6,18 @@ python:
 
 sudo: "required"
 
-before_install:
-  - "pip install virtualenv"
-  - "virtualenv venv"
-  - "source venv/bin/activate"
+env:
+  matrix:
+  - TOXENV=foxpuppet GUI=1
 
-install:
-  - "pip install -r requirements.txt "
+before_install:
+  - pip install mozdownload
+  - mozdownload --type=daily
+  - for f in *; do if [[ $f =~ (mozilla-central|firefox-52) ]]; then tar -xvf $f; fi; break; done
+  - if [ $TOXENV ]; then sudo pip install tox; fi
+  - if [[ $TOXENV == 'foxpuppet' ]]; then wget -O geckodriver.tar.gz https://github.com/mozilla/geckodriver/releases/download/v0.10.0/geckodriver-v0.10.0-linux64.tar.gz; fi
+  - if [[ $TOXENV == 'foxpuppet' ]]; then gunzip -c geckodriver.tar.gz | tar xopf -; fi
+  - if [[ $TOXENV == 'foxpuppet' ]]; then chmod +x geckodriver && sudo mv geckodriver /usr/local/bin; fi
```

Comment:
Remove all the `if` blocks because we'll always need GeckoDriver. We might not want it when we're running flake8 but I'd take a more readable configuration over the cost of downloading a small binary.